### PR TITLE
[JENKINS-60413] Guard against PRs from unknown forks

### PR DESCRIPTION
**Pull request by Dohbedoh**
* JENKINS issue(s):
   * [JENKINS-60413](https://issues.jenkins-ci.org/browse/JENKINS-60413)
* Description:
    * Indexing fails when there are open PRs from unknown forks (for example deleted forks)
* Documentation changes:
    * N/A
* Users/aliases to notify:
    * @bitwiseman @v1v

### CODE REVIEW

**Comment by Dohbedoh** [comment]
I am trying to reconcile both [JENKINS-58450](https://issues.jenkins-ci.org/browse/JENKINS-58450) and [JENKINS-41246](https://issues.jenkins-ci.org/browse/JENKINS-41246) by passing `this.sourceOwner`. This should fix the regression and be aligned with the previous behavior for deleted forks. I am open to suggestions though.

**Comment by Dohbedoh** [comment]
Tested that it fixes the indexing and that it is considered as an untrusted source:

```
Checking pull request #1937
    (not from a trusted source)
    Met criteria
```

File: src/main/java/org/jenkinsci/plugins/github_branch_source/PullRequestSCMHead.java [hunk 1]
```diff
@@ -92,9 +92,9 @@
             this.origin = SCMHeadOrigin.DEFAULT;
         } else {
             // if the forked repo name differs from the upstream repo name
-            this.origin = pr.getHead().getRepository().getName().equalsIgnoreCase(pr.getBase().getRepository().getName())
+            this.origin = pr.getBase().getRepository().getName().equalsIgnoreCase(sourceRepo)
```
Comment (aheritier) [inline src/main/java/org/jenkinsci/plugins/github_branch_source/PullRequestSCMHead.java]: Your test and the proposed code looks good to me but I am not sure to understand why in such situation `pr.getBase().getRepository()` isn't null ?

**Comment by aheritier** [comment]
(I removed my previous comment. This is the repo from head which is null not the one from base. It's ok for me

**Pull request merged**